cmd/cosign/cli/verify/verify_attestation.go

```diff
@@ -122,7 +121,7 @@ func (c *VerifyAttestationCommand) Exec(ctx context.Context, images []string) (e
 			return errors.Wrap(err, "loading certificate from reference")
 		}
 		if c.CertChain == "" {
-			co.SigVerifier, err = signature.LoadVerifier(cert.PublicKey, crypto.SHA256)
+			co.SigVerifier, err = signature.LoadECDSAVerifier(cert.PublicKey.(*ecdsa.PublicKey), crypto.SHA256)
```

Comment:
Umm, this seems a regression, as we will want to support ECDS and non ECDSA keys

Reply:
Oh snaps, that must have been a bad rebase, Thanks for catching this!!!

Reply:
This was added in https://github.com/sigstore/cosign/pull/1740

Reply:
Fun fact, I wanted to take a crack at actually combining these and verify.go flag handling as they are identical (see the name of my branch LOL ), but then went off the rails and did this instead. But I'll try to get to it :)